### ISSUE TITLE
Focus\viewport position issues on patterns pages

### ISSUE THREAD

**Issue by ad9242** [mi6/ic-design-system#938]
## Summary of the bug

On the patterns pages, there are a couple of issues with the content in view

## 🪜 How to reproduce

**issue 1:** 
1) go to [top nav layout](https://design.sis.gov.uk/patterns/navigation-patterns/top-navigation-layout#introduction)
2) click on the "Show code" button
3) the focused element (the "Hide code" button) is now off screen so, if using a mouse, you have to scroll all the way down to hide it again

**issue 2:**
1) go to [top nav layout](https://design.sis.gov.uk/patterns/navigation-patterns/top-navigation-layout#introduction)
2) click on the "Show code" button
3) scroll all the way down to so the "Hide code" button is visible
4) click the button
5) the page jumps to the bottom of the page

## 🧐 Expected behaviour

**Issue 1:**
ideally the focused element would not be off-screen. unsure if this is a accessibility issue, but it is certainly annoying

**Issue 2:**
The page should not scroll to the bottom and the viewport should show the code preview corresponding to the button clicked

Suggested changes:
Some very quick mock ups to move all buttons to the top of the preview. I'm sure someone more skilled could improve:

![image](https://github.com/mi6/ic-design-system/assets/116102637/29b3544e-bfb3-44b1-aa13-5c9c5a452dca)

![image](https://github.com/mi6/ic-design-system/assets/116102637/6bc519dc-a3ad-4275-9cbc-861487261d87)

![image](https://github.com/mi6/ic-design-system/assets/116102637/3e600017-0dd5-43b2-93b0-e9c3d00eb18f)

**Comment by MI6-255** [comment]
@GCHQ-Developer-094 to refine which design

**Comment by GCHQ-Developer-094** [comment]
Out of the three, the first mock-up looks to be the most logical from an IA point of view (inline with the two implementation options, with copy and stackblitz in the layer beneath)

**Comment by GCHQ-Developer-465** [comment]
This work now needs to put on hold due to conflicts with the ongoing StackBlitz work. Myself and @GCHQ-Developer-112 have agreed that this work can be continued after the StackBlitz work has been merged in. 

FYI - I have made changes locally on a branch [here](https://github.com/mi6/ic-design-system/commit/1112c95bc1287e4d8c1222f6a0d5362c6b541873), therefore this work can be picked up from anyone after the StackBlitz work has been merged in.
